src/Control/SimpleSearchField.php

```diff
@@ -0,0 +1,170 @@
+    {
+        $this->searchParameter = $name;
+
+        return $this;
+    }
+
+    /**
+     * Get the search parameter
+     *
+     * @return string
+     */
+    public function getSearchParameter(): string
+    {
+        return $this->searchParameter ?: 'q';
+    }
+
+    /**
+     * Set the suggestion url
+     *
+     * @param   Url $url
+     *
+     * @return  $this
+     */
+    public function setSuggestionUrl(Url $url): self
+    {
+        $this->suggestionUrl = $url;
+
+        return $this;
+    }
+
+    /**
+     * Get the suggestion url
+     *
+     * @return Url
+     */
+    public function getSuggestionUrl(): Url
+    {
+        return $this->suggestionUrl;
+    }
+
+    /**
+     * Set submit label
+     *
+     * @param string $label
+     *
+     * @return  $this
+     */
+    public function setSubmitLabel(string $label): self
+    {
+        $this->submitLabel = $label;
+
+        return $this;
+    }
+
+    /**
+     * Get submit label
+     *
+     * @return string
+     */
+    public function getSubmitLabel(): string
+    {
+        return $this->submitLabel ?? t('Submit');
+    }
+
+    public function assemble()
+    {
+        $filterInput = new InputElement(null, [
+            'type'                      => 'text',
+            'placeholder'               => 'Type to search',
+            'class'                     => 'search-input',
+            'id'                        => 'search-input',
+            'autocomplete'              => 'off',
+            'required'                  => true,
+            'data-no-auto-submit'       => true,
+            'data-no-js-placeholder'    => true,
+            'data-enrichment-type'      => 'terms',
+            'data-term-separator'       => self::TERM_SEPARATOR,
+            'data-term-mode'            => 'read-only',
+            'data-term-direction'       => 'vertical',
+            'data-data-input'           => '#data-input',
+            'data-term-input'           => '#term-input',
+            'data-term-container'       => '#term-container',
+            'data-term-suggestions'     => '#term-suggestions',
+            'data-suggest-url'          => $this->getSuggestionUrl()
+        ]);
+
+        $dataInput = new InputElement('data', ['type' => 'hidden', 'id' => 'data-input']);
+
+        $termInput = new InputElement($this->getSearchParameter(), ['type' => 'hidden', 'id' => 'term-input']);
+        $this->registerElement($termInput);
+
+        $termContainer = new HtmlElement(
+            'div',
+            Attributes::create(['id' => 'term-container', 'class' => 'term-container'])
+        );
+
+        $termSuggestions = new HtmlElement(
+            'div',
+            Attributes::create(['id' => 'term-suggestions', 'class' => 'search-suggestions'])
+        );
+
+        $submitButton = new SubmitElement(
+            'submit',
+            ['label' => $this->getSubmitLabel(), 'class' => 'btn-primary']
```

Comment:
`'btn-primary'` is from Icinga Web 2 as well. You shouldn't need it anyway, you have your own style for it.